### DISCUSSION TITLE
It seems that the function "if(expr1,expr2,expr3)" is not supported。

### DISCUSSION THREAD

**Discussion by GaryBoBo** [JSQLParser/JSqlParser#1277] [Q&A] [answered]
Encountered unexpected token: "(" "("

**Comment by manticore-projects** [comment, marked as answer]
Should be supported with 4.2-Snapshot.
Please verify and provide a full test case otherwise.

**Reply by GaryBoBo** [reply]
SELECT if(1 in (1,2,3),'yes','no') from dual

**Reply by manticore-projects** [reply]
`1 in (1,2,3)` should not be a valid expression.

**Reply by manticore-projects** [reply]
But it parses correctly. You can [try here online](http://jsqlformatter.manticore-projects.com/jsqlformatter/demo.html?args=-c%20MoUQMiDCAqAECSAzAFLAjAgcrVaA0sATAQMywCUBA5AJ4CmAzldQHYD2VFAUAGIBKAeQCysACYBXAIYAbLgG4gA+).

**Reply by GaryBoBo** [reply]
Use CCJSqlParserUtil.parse(String sql) to parse this sql  throws exception.

**Comment by GaryBoBo** [comment]
Thank you! This [JSQLParser Snapshot](https://github.com/manticore-projects/jsqlformatter/releases/download/0.1.6/jsqlparser-4.1-SNAPSHOT.jar) solved my problem.

**Reply by manticore-projects** [reply]
Welcome, you can also use the JSQLParser 4.2 Snapshot from upstream.
The nonofficial patched JSQLParser 4.1 Snapshot was a temporary Bridge Gap Solution and is no longer needed. 

Please mark this discussion as `ANSWERED`, thank you.

**Reply by manticore-projects** [reply]
Please refer to issue  #1284 which has been resolved in the meantime by PR #1285.
After that PR you can definitely use upstream.
